CODING_STYLE.md

```diff
@@ -46,6 +46,8 @@ foo->bar(someLongVariableName,
 cout << "some very long string that contains completely irrelevant text that talks about this and that and contains the words \"lorem\" and \"ipsum\"" << endl;
 ```
 
+To assist in getting the right coding style automatically, an [EditorConfig](https://editorconfig.org/) [`.editorconfig`](https://github.com/ethereum/solidity/blob/develop/.editorconfig) file is in the repository which describes some of the style used. Many IDE's and editors will notice and follow this automatically.
```

Comment:
Do people really want to read this here if it is automatic?

Reply:
Actually, it was _not_ automatic for me. I use GNU Emacs, and there is an additional thing that needs to be done to tell it to read editorconfig.

Reply:
The `.editorconfig` basically just tells you to use tabs for C++ files. A proper editor could deduce that from the file itself (unless a new file is created). So perhaps it would be better to remove `.editorconfig` altogether?

Reply:
For whatevere reason Emacs didn't grok this automatically. 

(Emacs does allow for a lot of control, it just uses a different convention.)

The thing about `.editorconfig` is that it make things explicit in a code-like way. Isn't _explicit_ better than _implcit_?

Reply:
Without `.editorconfig`, github will show tab as 8 spaces which is not desired (ref. #4600). Also Vim cannot automatically determine to use tabs instead of spaces.

Reply:
I think this change could even split out into a separate PR. In general I'm not in favor of adding IDE / editor specific files to the repository. As far as I can tell I would not be picked up by e.g. Visual Studio Code or QtCreator without additional plugins.

Reply:
In the long term, if I work on this project I can make a stab at writing a separate doc just for developers of solidity and that can talk about the differenent IDEs, debuggers, editorrs, evironments that people use and what might be adjusted for best use in each.

Reply:
Some of those aspects are already included in `docs/contributing.rst`. I'd say we could think about how to streamline both documents. If you really want to have the `.editorconfig` mentioned here, I'm fine with including this paragraph.